slam-libraries/viam-cartographer/src/mapping/map_builder.cc

```diff
@@ -81,6 +81,23 @@ void MapBuilder::SaveMapToFile(bool include_unfinished_submaps,
     }
 }
 
+std::string MapBuilder::SaveMapToStream(const std::string path_to_dir) {
+    std::string filename = path_to_dir + "/" + "temp.pbstream";
+    SaveMapToFile(false, filename);
```

Comment:
Is it possible for this to fail?

Reply:
`SaveMapToFile` cannot fail as the failure from `SerializeToFile` is checked in that function. Failure in the `SerializeStateToFile` can only come from the closing of the `writer` (which closes out the file being written to).

Reply:
I think we should change SaveMapToFile to return a boolean indicating whether it succeeded. If it failed, then GetInternalState should return an error.

Reply:
What about the case where Cartographer started writing to the file, then failed half way through, resuling in corrupt data, which we then read & return to the end user? Shouldn't we return a GRPC error if carto failed to write all its state to disk?